lib/gift/bloc/redeem_gift_bloc.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:ardrive/authentication/ardrive_auth.dart';
+import 'package:ardrive/turbo/services/payment_service.dart';
+import 'package:ardrive/utils/logger/logger.dart';
+import 'package:equatable/equatable.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+part 'redeem_gift_event.dart';
+part 'redeem_gift_state.dart';
+
+class RedeemGiftBloc extends Bloc<RedeemGiftEvent, RedeemGiftState> {
+  final PaymentService _paymentService;
+  final ArDriveAuth _auth;
+
+  RedeemGiftBloc({
+    required PaymentService paymentService,
+    required ArDriveAuth auth,
+  })  : _auth = auth,
+        _paymentService = paymentService,
+        super(RedeemGiftInitial()) {
+    on<RedeemGiftEvent>((event, emit) async {
+      if (event is RedeemGiftLoad) {
+        await _handleRedeemGiftLoad(emit, event);
+      }
+    });
```

Comment:
No else? If there's no other event we can directly set the callback for the event explicitly.
```suggestion
    on<RedeemGiftLoad>(_handleRedeemGiftLoad);
```